great_expectations/checkpoint/checkpoint.py

```diff
@@ -1378,6 +1384,7 @@ class SimpleCheckpoint(Checkpoint):
     """
 
     _configurator_class = SimpleCheckpointConfigurator
+    name: str
```

Comment:
Is this `name` needed?  By the way, the "Configurator" is a very complex solution to the problem that was not posed correctly.

Reply:
Yes, otherwise `mypy` thinks the `self.name` attribute could be `None`, but it's always present.